core/src/main/java/com/google/android/fhir/db/impl/dao/LocalChangeUtils.kt

```diff
@@ -85,16 +87,30 @@ object LocalChangeUtils {
     }
 
     /**
-     * Merge two JSON patch strings by concatenating their elements into a new JSON array.
+     * Merge two JSON patch strings by concatenating their elements, which should JSON patch
+     * operations, into a new JSON array.
+     * For duplicate operations, the operation from second patch is picked.
      */
     @JvmStatic
     private fun mergePatches(firstPatch: String, secondPatch: String): String {
-        val objectMapper = ObjectMapper()
         // TODO: validate patches are RFC 6902 compliant JSON patches
-        val first = objectMapper.readValue(firstPatch, JsonPatch::class.java)
-        val updater = objectMapper.readerForUpdating(first)
-        val merged = updater.readValue(secondPatch, JsonPatch::class.java)
-        return merged.toString()
+        val firstPatchJsonArray = JSONArray(firstPatch)
+        val secondPatchJsonArray = JSONArray(secondPatch)
+        val mergeMap = HashMap<Pair<String, String>, JSONObject>()
+
+        for (i in 0 until firstPatchJsonArray.length()) {
+            val operation = firstPatchJsonArray.optJSONObject(i) ?: JSONObject()
+            val key = operation.optString("op") to operation.optString("path")
+            mergeMap[key] = operation
+        }
```

Comment:
I think you can do something along these lines:

```suggestion
        val firstJsonObjectList = (0..firstJsonArray.length()).map { firstJsonArray.optJSONObject(it) }
        val firstJsonObjectMap = firstJsonObjectList.associate {
            it.optString("op") to it
        }.toMutableMap()
        
        val secondJsonObjectList = (0..secondJsonArray.length()).map { secondJsonObjectList.optJSONObject(it) }
        val secondJsonObjectMap = secondJsonObjectList.associate {
            it.optString("op") to it
        }.toMap()
        
        firstJsonObjectMap.putAll(secondJsonObjectList)
```

also may create a utility function for the repeated code? up to you.

Reply:
Done.